src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncProgressResolver.cs

```diff
@@ -83,32 +83,18 @@ public long FindBestFullState()
             return _blockTree.FindHeader(blockHash)?.TotalDifficulty == 0 ? null : _blockTree.FindHeader(blockHash)?.TotalDifficulty;
         }
 
-        public bool IsFastBlocksHeadersFinished() => !IsFastBlocks() || (!_syncConfig.DownloadHeadersInFastSync ||
-                                                                         (_headersSyncFeed?.IsFinished == true));
+        public bool IsFastBlocksHeadersFinished() => !IsFastBlocks() || !_syncConfig.DownloadHeadersInFastSync || _headersSyncFeed?.IsFinished == true;
 
-        public bool IsFastBlocksBodiesFinished() => !IsFastBlocks() || (!_syncConfig.DownloadBodiesInFastSync ||
-                                                                        (_bodiesSyncFeed?.IsFinished == true));
+        public bool IsFastBlocksBodiesFinished() => !IsFastBlocks() || !_syncConfig.DownloadBodiesInFastSync || _bodiesSyncFeed?.IsFinished == true;
 
-        public bool IsFastBlocksReceiptsFinished() => !IsFastBlocks() || (!_syncConfig.DownloadReceiptsInFastSync ||
-                                                                          (_receiptsSyncFeed?.IsFinished == true));
+        public bool IsFastBlocksReceiptsFinished() => !IsFastBlocks() || !_syncConfig.DownloadReceiptsInFastSync || _receiptsSyncFeed?.IsFinished == true;
 
         public bool IsSnapGetRangesFinished() => _snapSyncFeed?.IsFinished ?? true;
 
         public void RecalculateProgressPointers() => _blockTree.RecalculateTreeLevels();
 
-        private bool IsFastBlocks()
-        {
-            // maybe eliminate the entire code refeferencing this function why? fast block depends on fastsync so this code might just be check that is now unnecessay!
-            // name isFastBlocks left in place for correlation with protocol specification.
-            bool isFastBlocks = _syncConfig.FastSync; 
+        private bool IsFastBlocks() => _syncConfig.FastSync && _syncConfig.PivotNumberParsed != 0L; // if pivot number is 0 then it is equivalent to fast blocks disabled
```

Comment:
@LukaszRozmej 
Why is does this line work.
Seems to be the only change in logic within the fixes

Reply:
its just logic simplification - works exactly the same